src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.cinema.dao.impl;
+
+import com.cinema.dao.MovieSessionDao;
+import com.cinema.exeption.DataProcessingException;
+import com.cinema.lib.Dao;
+import com.cinema.model.MovieSession;
+import com.cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllMovieSessionQuery
+                    = session.createQuery("FROM MovieSession WHERE movie.id = :id "
+                    + "AND showTime BETWEEN :start AND :end", MovieSession.class);
+            getAllMovieSessionQuery.setParameter("id", movieId);
+            getAllMovieSessionQuery.setParameter("start", date.atTime(LocalTime.MIN));
+            getAllMovieSessionQuery.setParameter("end", date.atTime(LocalTime.MAX));
+            return getAllMovieSessionQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all movie session", e);
```

Comment:
and this message

Reply:
I don't know what can I add to this message

Reply:
> I don't know what can I add to this message

you can add info about input parameters (we're searching sessions for particular movie and date)

Reply:
Done